kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -378,37 +383,38 @@ impl<'tcx> GotocCtx<'tcx> {
     /// The otherwise value is stores as the last value of targets.
     fn codegen_switch_int(
         &mut self,
-        discr: &Operand<'tcx>,
+        discr: &Operand,
         targets: &SwitchTargets,
         loc: Location,
     ) -> Stmt {
-        let v = self.codegen_operand(discr);
+        let v = self.codegen_operand_stable(discr);
         let switch_ty = v.typ().clone();
-        if targets.all_targets().len() == 1 {
```

Comment:
Not sure what's happening here, but it can have a significant impact. Did the cases get swapped?

Reply:
No. This was an incorrect optimization on our side. Before this PR, the `if` block was unreachable.

Following the comments in our old code, I believe that we were assuming that `all_targets()` did not include the `otherwise` target, (aka default target), but in does. So a SwitchTarget here always has at least 2 targets. With 2 targets, the switch behaves like a if then else.

Reply:
This was a bit confusing because `SwitchTargets::all_targets()` includes the `otherwise` target, but the `SwithTargets::iter()` did not.

That's why in StableMIR we use `branches` for targets that are conditional ones.

Reply:
Thanks for the explanation! I don't see anything wrong in the new version, and since there are no regression failures (including the performance benchmarks) we can accept it.

Reply:
@celinval feel free to close https://github.com/model-checking/kani/issues/2712 if it's fixed by this change.